FestivalPlanner/films/tests.py

```diff
@@ -287,30 +319,184 @@ def test_rating_of_created_film_without_login(self):
         when not logged in.
         """
         # Arrange.
-        created_film = create_film(film_id=6000, title='New Adventure', minutes=114)
-        url = reverse('films:rating', args=(created_film.film_id,))
+        saved_film = create_film(film_id=6000, title='New Adventure', minutes=114)
+        url = reverse('films:results', args=[saved_film.pk])
 
         # Act.
         response = self.client.get(url)
 
         # Assert.
         self.assertEqual(response.status_code, HTTPStatus.FOUND)
+        with self.assertRaisesRegex(AssertionError, r"^302 != 200 : Couldn't retrieve content"):
+            self.assertContains(response, saved_film.title)
+
+    def test_created_film_unrated_logged_in(self):
+        """
+        The rating view with film id of a film that was newly added to
+        the database displays no ratings for the given film.
+        """
+        # Arrange.
+        fan = self.regular_fan
+        _ = self.get_regular_fan_request()
+        fan_post_data = {'selected_fan': [fan.name]}
+        fan_response = self.client.post(reverse('films:film_fan'), data=fan_post_data)
+
+        saved_film = create_film(film_id=6001, title='New Adventures', minutes=115)
+
+        # Act.
+        post_response = self.client.get(reverse('films:results', args=[saved_film.pk]))
+
+        # Assert.
+        self.assertEqual(fan_response.status_code, HTTPStatus.FOUND)
+        self.assertEqual(post_response.status_code, HTTPStatus.OK)
+        self.assertContains(post_response, saved_film.title)
+        self.assert_fan_row(fan, UNRATED_STR, post_response)
 
     def test_rating_of_created_film_logged_in(self):
         """
         The rating view with film id of a film that was newly added to
         the database displays the ratings of the given film.
         """
         # Arrange.
-        created_film = create_film(film_id=6001, title='New Adventures', minutes=115)
-        request = self.get_regular_fan_request()
+        fan = self.regular_fan
+        _ = self.get_regular_fan_request()
+        fan_post_data = {'selected_fan': [fan.name]}
+        fan_response = self.client.post(reverse('films:film_fan'), data=fan_post_data)
```

Comment:
Please DRY and create a method for this (or integrate it in `get_regular_fan_request()` / `get_admin_request()`.

Reply:
Done.